exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/AvgWidthMergedStatistic.java

```diff
@@ -0,0 +1,135 @@
+    assert (input.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    MapVector inputMap = (MapVector) input;
+    for (ValueVector vv : inputMap) {
+      String colName = vv.getField().getLastName();
+      NullableFloat8Holder colSumHolder;
+      if (sumHolder.get(colName) != null) {
+        colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      } else {
+        colSumHolder = new NullableFloat8Holder();
+        sumHolder.put(colName, colSumHolder);
+      }
+      Object val = vv.getAccessor().getObject(0);
+      if (val != null) {
+        colSumHolder.value += (double) val;
+        colSumHolder.isSet = 1;
+      }
+    }
+  }
+
+  @Override
+  public Object getStat(String colName) {
+      if (mergeComplete != true) {
+        throw new IllegalStateException(
+            String.format("Statistic `%s` has not completed merging statistics", name));
+      }
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      return (long) (colSumHolder.value/ getRowCount(colName));
+    }
+
+  @Override
+  public void setOutput(ValueVector output) {
+    // Check the input is a Map Vector
+    assert (output.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    // Dependencies have been configured correctly
+    assert (configureComplete == true);
+    MapVector outputMap = (MapVector) output;
+
+    for (ValueVector outMapCol : outputMap) {
+      String colName = outMapCol.getField().getLastName();
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      NullableFloat8Vector vv = (NullableFloat8Vector) outMapCol;
+      vv.allocateNewSafe();
+      vv.getMutator().setSafe(0, (colSumHolder.value / getRowCount(colName)));
+    }
+    mergeComplete = true;
+  }
+
+  @Override
+  public void configure(Object configurations) {
+    List<MergedStatistic> statistics = (List<MergedStatistic>) configurations;
+    for (MergedStatistic statistic : statistics) {
+      if (statistic.getName().equals("type")) {
+        types = statistic;
+      } else if (statistic.getName().equals("statcount")) {
+        statCounts = statistic;
+      } else if (statistic.getName().equals("nonnullstatcount")) {
+        nonNullStatCounts = statistic;
+      }
+    }
+    assert (types != null && statCounts != null && nonNullStatCounts != null);
+    configureComplete = true;
+  }
+
+  private long getRowCount(String colName) {
+    int type = (int) types.getStat(colName);
+    // If variable type - then use the nonNullCount. Otherwise, use the Count,
+    // since even NULL values take up the same space.
+    if (type == TypeProtos.MinorType.VAR16CHAR.getNumber()
+        || type == TypeProtos.MinorType.VARCHAR.getNumber()
+        || type == TypeProtos.MinorType.VARBINARY.getNumber()) {
+      return (long) nonNullStatCounts.getStat(colName);
+    } else {
+      return (long) statCounts.getStat(colName);
```

Comment:
If `getStat` is typed, as noted above, then:
```
return statCounts.getStat(colName);
```

I do wonder, however if the structure can be simplified. We do many hash lookups here, but they all seem unnecessary. We have a list of known list of columns. Can columns have an index that corresponds to the map vector items, and the various collections used here? Hash map lookups are not free; doing many of them unnecessarily impacts performance.